src/jit/simd.cpp

```diff
@@ -129,236 +129,494 @@ var_types Compiler::getBaseTypeAndSizeOfSIMDType(CORINFO_CLASS_HANDLE typeHnd, u
+        }
+        else if (typeHnd == Vector256ULongHandle)
+        {
+            simdBaseType = TYP_ULONG;
+            size         = YMM_REGSIZE_BYTES;
+            JITDUMP("  Known type Vector256<ulong>\n");
+        }
+        else if (typeHnd == Vector256FloatHandle)
+        {
+            simdBaseType = TYP_FLOAT;
+            size         = YMM_REGSIZE_BYTES;
+            JITDUMP("  Known type Vector128<float>\n");
+        }
+        else if (typeHnd == Vector128DoubleHandle)
+        {
+            simdBaseType = TYP_DOUBLE;
+            size         = XMM_REGSIZE_BYTES;
+            JITDUMP("  Known type Vector128<double>\n");
+        }
+        else if (typeHnd == Vector128IntHandle)
+        {
+            simdBaseType = TYP_INT;
+            size         = XMM_REGSIZE_BYTES;
+            JITDUMP("  Known type Vector128<int>\n");
+        }
+        else if (typeHnd == Vector128UIntHandle)
+        {
+            simdBaseType = TYP_UINT;
+            size         = XMM_REGSIZE_BYTES;
+            JITDUMP("  Known type Vector128<uint>\n");
+        }
+        else if (typeHnd == Vector128ShortHandle)
+        {
+            simdBaseType = TYP_SHORT;
+            size         = XMM_REGSIZE_BYTES;
+            JITDUMP("  Known type Vector128<short>\n");
+        }
+        else if (typeHnd == Vector128UShortHandle)
+        {
+            simdBaseType = TYP_CHAR; //TODO TYP_USHORT;
+            size         = XMM_REGSIZE_BYTES;
+            JITDUMP("  Known type Vector128<ushort>\n");
+        }
+        else if (typeHnd == Vector128ByteHandle)
+        {
+            simdBaseType = TYP_BYTE;
+            size         = XMM_REGSIZE_BYTES;
+            JITDUMP("  Known type Vector128<sbyte>\n");
+        }
+        else if (typeHnd == Vector128UByteHandle)
+        {
+            simdBaseType = TYP_UBYTE;
+            size         = XMM_REGSIZE_BYTES;
+            JITDUMP("  Known type Vector128<byte>\n");
+        }
+        else if (typeHnd == Vector128LongHandle)
+        {
+            simdBaseType = TYP_LONG;
+            size         = XMM_REGSIZE_BYTES;
+            JITDUMP("  Known type Vector128<long>\n");
+        }
+        else if (typeHnd == Vector128ULongHandle)
+        {
+            simdBaseType = TYP_ULONG;
+            size         = XMM_REGSIZE_BYTES;
+            JITDUMP("  Known type Vector128<ulong>\n");
         }
 
-        *sizeBytes = size;
-        setUsesSIMDTypes(true);
+        // slow path search
+        if (simdBaseType == TYP_UNKNOWN)
+        {
+            // Doesn't match with any of the cached type handles.
+            const char*          className   = getClassNameFromMetadata(typeHnd, nullptr);
+            CORINFO_CLASS_HANDLE baseTypeHnd = getTypeInstantiationArgument(typeHnd, 0);
+
+            if (baseTypeHnd != nullptr)
+            {
+                CorInfoType type = info.compCompHnd->getTypeForPrimitivNumericClass(baseTypeHnd);
```

Comment:
@AndyAyersMS `getTypeForPrimitivNumericClass` is called here. That converts the return value of `getTypeInstantiationArgument` to `CorInfoType`.

Reply:
Ah, GitHub was hiding the diff. Thanks.